### PR TITLE
COR-674: Update Fishery to support bulk operations

### PR DESCRIPTION
**Pull request by RiyaAggarwal123**

### CODE REVIEW

File: lib/__tests__/factory.test.ts [hunk 1]
```diff
@@ -154,6 +188,35 @@ describe('onCreate', () => {
...
+
+    const bulkUserFactory = userFactory.onBulkCreate(onBulkCreateFn);
+
+    const promise = bulkUserFactory.createList(2, { name: 'susan' }, { bulkCreate: true });
+    expect(promise).toBeInstanceOf(Promise);
+
+    const users = await promise;
+
+    expect(users.every(u => u.name === 'Bill')).toBeTruthy();
+    expect(onBulkCreateFn).toHaveBeenCalledTimes(1);
```
Comment (dhirendra-linq) [inline lib/__tests__/factory.test.ts:223]: Should we also expect `calledWith` here?

File: lib/__tests__/factory.test.ts [hunk 2]
```diff
@@ -96,6 +96,40 @@ describe('factory.createList', () => {
...
+  });
+
+  it('does not calls onBulkCreate function if bulkCreate value passed false in options', async () => {
```
Comment (dhirendra-linq) [inline lib/__tests__/factory.test.ts]: I think this gets tested along with `calls onCreate for each item`, no need to duplicate the test.   You can add `expect(onBulkCreateFn).toHaveBeenCalledTimes(0);` in that test case.

File: lib/__tests__/factory.test.ts [hunk 3]
```diff
@@ -96,6 +96,40 @@ describe('factory.createList', () => {
...
+
+    const bulkUserFactory = userFactory.onBulkCreate(onBulkCreateFn);
+
+    const promise = bulkUserFactory.createList(2, { name: 'susan' }, { bulkCreate: true });
+    expect(promise).toBeInstanceOf(Promise);
+
+    const users = await promise;
+
+    expect(users.every(u => u.name === 'Bill')).toBeTruthy();
+    expect(onBulkCreateFn).toHaveBeenCalledTimes(1);
```
Comment (dhirendra-linq) [inline lib/__tests__/factory.test.ts:127]: Let's add `expect(onCreateFn).toHaveBeenCalledTimes(0);` here.

File: lib/__tests__/factory.test.ts [hunk 4]
```diff
@@ -96,6 +96,40 @@ describe('factory.createList', () => {
...
     expect(onCreateFn).toHaveBeenCalledTimes(2);
   });
+
+  it('calls onBulkCreate function if bulkCreate value passed true in options', async () => {
+    const onBulkCreateFn = jest.fn(async users => {
+      users.every((user: { name: string; }) => user.name = 'Bill')
+      return Promise.resolve(users);
+    });
+
+    const bulkUserFactory = userFactory.onBulkCreate(onBulkCreateFn);
```
Comment (dhirendra-linq) [inline lib/__tests__/factory.test.ts]: Build factory with both `onCreate` and `onBulkCreate` hooks.

File: lib/__tests__/factory.test.ts (hunk 2 above)
Comment (RiyaAggarwal123) [reply lib/__tests__/factory.test.ts]: @dhirendrx in that case we cannot `expect(onBulkCreateFn) ` till that we should have definition for it

**Review by dhirendra-linq — APPROVED** [review]

**Comment by dhirendra-linq** [comment]
@RiyaAggarwal123 Rename this package to `@roundtrip/fishery` in `package.json`.

**Pull request merged**